### ISSUE TITLE
[Bug] Codespell doesn't offer to capitalize word result if it doesn't start with the same first letter (?)

### ISSUE THREAD

**Issue by luzpaz** [codespell-project/codespell#1883]
`./src/effects/Reverb.cpp:455: Rever ==> Revert, fever`  
In the analysis example above `fever` is not capitalized

**Comment by peternewman** [comment]
I can't see this in the dictionary @luzpaz . Is it personal custom dictionary?

**Comment by luzpaz** [comment]
No... this is not a custom dictionary. And I'm not referring to how the words are displayed in the custom dictionary. This is specifically related to the logic that is used by codespell to capitalize/capital-case the suggested correctly-spelled words because the typo itself was capitalized/capital-cased. Make sense?

**Comment by peternewman** [comment]
Ah, which version are you running? I tried various searches via GitHub and failed to find it, but by inspecting the dictionary:
https://github.com/codespell-project/codespell/blob/f22aa4ae236de55d709defe6f7022c15cd787221/codespell_lib/data/dictionary.txt#L24590

Please can you also share the whole line it's matching on and any command line flags/config you're using.

Also hopefully there's a PR to add reverb as a suggestion?

**Comment by peternewman** [comment]
Edit, agreed, it's broken:
```
text.txt:3: rever ==> revert, refer, fever
text.txt:4: Rever ==> Revert, refer, fever
text.txt:5: REVER ==> REVERT, REFER, FEVER
```
fix_case needs erm fixing. Well some tests adding first. It needs a for each around each suggestion individually, but also a quick check the reason isn't getting passed to it if present:
https://github.com/codespell-project/codespell/blob/079b649d9ba381c0cc7a62f5f88d043e5d81c5ea/codespell_lib/_codespell.py#L471-L478

**Comment by bl-ue** [comment]
Ah, I see @peternewman: the problem is that `fix_case` is being called essentially like this:
```python
fix_case("Rever", "revert, refer, fever")
```
So it's capitalizing `fixword`, which is why only the first word is capitalized.

**Comment by peternewman** [comment]
Exactly. But the fix is not to just do a naive split, as the last one shouldn't be capitalised because it's a potential note.

**Comment by bl-ue** [comment]
Oh, that's a good point.  So maybe we should change `build_dict()` and `Misspelling` to have a list of suggestions and an optional comment?  That sounds like the best idea to me.

https://github.com/codespell-project/codespell/blob/master/codespell_lib/_codespell.py#L454-L479

I could easily undertake it.

**Comment by peternewman** [comment]
I think that handles it already, and `reason` is the reason. Basically just make sure the tests cover both cases, but yeah, feel free to get stuck in.

**Comment by peternewman** [comment]
@DimitriPapadopoulos fixed this in #2223

> So maybe we should change `build_dict()` and `Misspelling` to have a list of suggestions and an optional comment? That sounds like the best idea to me.
> 
> https://github.com/codespell-project/codespell/blob/7db90e08164efbf642fbc2724174791d46fa0f73/codespell_lib/_codespell.py#L454-L479
> 
> I could easily undertake it.

But something like this would be good still @bl-ue !
